### DISCUSSION TITLE
Making text optional if a placeholder is empty

### DISCUSSION THREAD

**Discussion by ant54321** [jgraph/drawio-desktop#1827] [Q&A] [answered]
I am using a csv to help automate drawing a family tree using Arrange/Insert/Advanced/CSV. I have been following the example file but have hit a wall with the following problem.

I have placeholders for MarriedTo, MarriedDate and MarriedLocation.  I want text to appear in my shape before these placeholders so that a line in the shape would appear something like this:

"Married to %placeholder1% on %placeholder2% in %placeholder3%"

I've managed to do that using the following line in my CSV, but of course if the person in the Shape isn't married, that appears as

"Married to on at"

# label: <span style="color:black;"> <font style="font-size: 13px;">%Name%</font></span><br><span style="color:navy;"><span style="color:maroon;">Married %Married% on %DoM% in %Mlocation%</span>

Is there  a way of making text optional so that it only appears if there is text in the placeholder?

**Comment by alderg** [comment]
This is currently not possible. As a workaround, try setting `noLabel=1` in the cell style if those placeholders are empty.

**Reply by ant54321** [reply]
Thanks for your reply alderg. I'm new to Drawio, and dont know html or json (but can muddle through with help) and would appreciate you expanding on your suggestion. How do I set nolabel = 1? I don't know how to find the cell style for a placeholder I've found the style for the shape and set it in there as a test but it made no difference. I'm thinking I would want to do that somehow in the csv file and not in the diagram itself as the family tree has many people, but again how would I know the placeholder is empty in that csv? 

There are lots of comments in the csv example about things I haven't touched, because I don't understand the comments. I was hoping something like using vars (see below) might help but have no idea because there isn't an example shown on how to use them.

## JSON for variables   in styles of the form {"name": "value", "name":   "value"} where name is a string
--
## that will replace a placeholder in a   style.
#
# vars: -
#

**Comment by ant54321** [comment, marked as answer]
I've just realised that I can use an excel IF to add the text "Married " to the placeholder text if the placeholder text isn't blank in my spreadsheet that I'm using to build my CSV data. If someone has a better approach to what I'm trying to do I'd be pleased to hear it.
